charms/worker/k8s/src/charm.py

```diff
@@ -317,6 +317,12 @@ def _check_k8sd_ready(self):
         status.add(ops.MaintenanceStatus("Ensuring snap readiness"))
         self.api_manager.check_k8sd_ready()
 
+    def _get_extra_sans(self):
+        """Retrieve the certificate extra SANs."""
+        extra_sans_str = str(self.config.get("kube-apiserver-extra-sans") or "")
+        configured_sans = [san.strip(" ") for san in extra_sans_str.split(",") if san.strip(" ")]
+        return [_get_public_address()] + configured_sans
```

Comment:
prevent duplicate sans being configured

```suggestion
        all_sans = set(extra_sans_str.strip().split()) | {_get_public_address()}
        return sorted(all_sans)
```

Reply:
Good point.